### PR TITLE
[tools] Add a script that can update a PR's CI status on GitHub.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
Salient points:
_Status_ can be manually specified or derived from the output of `Run-UnitTests` via _TestXMLDirectory_.
_Description_ is optional, and will be populated with test run results.
_PullRequest_ must take the form `REMOTE BRANCHNAME`
_Repository_ must take the form `USERNAME/REPONAME`

### CODE REVIEW

**Comment by rajsesh-msft** [comment]
This is awesome, thanks!

File: tools/build/Submit-GithubBuildStatus.ps1 [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	If ($disabledCount -Gt 0) {
+		$explanation += " ({0} skipped)" -f $disabledCount
+	}
+	If ($testCount -Gt 0 -And $failureCount -Gt 0) {
+		$Status = "failure"
+		$explanation = "{1}/{0} tests failed" -f $testCount, $failureCount
+		If ($disabledCount -Gt 0) {
+			$explanation += " ({0} skipped)" -f $disabledCount
+		}
+	} ElseIf ($testCount -Eq 0) {
```
Comment (rajsesh-msft) [inline tools/build/Submit-GithubBuildStatus.ps1]: may be better to move this up front to avoid the -And condition above.

**Comment by msft-Jeyaram** [comment]
:fire:

**Comment by rajsesh-msft** [comment]
:shipit:

**Comment by rajsesh-msft** [comment]
In CI build, hopefully last time this is manually updated. :8ball:

**Pull request merged**